src/mapper.rs

```diff
@@ -79,9 +87,16 @@ fn iterate_with_lines<'a>(
         } else {
             member.original_startline + frame.line - member.startline
         };
-        // when an inlined function is from a foreign class, we
-        // don’t know the file it is defined in.
-        let file = if member.original_class.is_some() {
+        let file = if member.original_file.is_some() {
+            let file_name = member.original_file.unwrap();
+            if file_name.eq("R8$$SyntheticClass") {
+                extract_class_name(member.original_class.unwrap_or(frame.class))
```

Comment:
Does this fallback to `frame.class` make sense to you? I still try to really understand how `original_class` is inflated: e.g. in the text I provided `test_remap_source_file`, `original_class` is empty for `at io.wzieba.r8fullmoderenamessources.MainActivity$$ExternalSyntheticLambda0.onClick`, hence the fallback.